src/coreclr/jit/fgopt.cpp

```diff
@@ -3412,9 +3412,24 @@ bool Compiler::fgReorderBlocks(bool useProfile)
         }
     }
 
-    // If we will be reordering blocks, ensure the false target of a BBJ_COND block is its next block
     if (useProfile)
     {
+        if (JitConfig.JitDoReversePostOrderLayout())
+        {
+            fgDoReversePostOrderLayout();
+            fgMoveColdBlocks();
+
+#ifdef DEBUG
+            if (expensiveDebugCheckLevel >= 2)
+            {
+                fgDebugCheckBBlist();
+            }
+#endif // DEBUG
+
+            return true;
```

Comment:
I assume it's unlikely that we won't move blocks here? The other variant tries to keep track, but I'm not sure it is worth the extra bookkeeping. All we are doing by sometimes returning false is speeding up the checked jit a tiny bit.

Reply:
That was my reasoning for hard-coding the return value.